src/core/agent/services/connectionService.ts

```diff
@@ -475,6 +481,14 @@ class ConnectionService extends AgentService {
     const pendingDeletions = await this.getConnectionsPendingDeletion();
     for (const id of pendingDeletions) {
       await this.deleteConnectionById(id);
+      this.markAgentStatus(true);
```

Comment:
Why would we mark the agent status as online here? Please read my comments more carefully.

I said we should be calling `removeConnectionsPendingDeletion` etc once the agent is online, so that call moves within the function called `markAgentStatus`.